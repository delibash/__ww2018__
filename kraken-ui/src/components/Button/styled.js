import styled from 'styled-components';
import { colors } from './../../global/constants';

const StyledButton = styled.button`
  padding: .8rem 1.8rem;
  text-transform: capitalize;
  font-weight: 300;
  font-size: 1.3rem;
  border: none;
  border-radius: .2rem;
  background-color: ${props => !props.disabled ? colors.BLUE : colors.LIGHTER_GRAY};
  color: ${props => !props.disabled ? 'white' : colors.BLACK};
  cursor: ${props => !props.disabled ? 'pointer' : 'default'};
  &:hover {
    background-color: ${colors.LIGHTER_GRAY};
    color: ${colors.BLACK}
  }
  `;

export {
  StyledButton
};