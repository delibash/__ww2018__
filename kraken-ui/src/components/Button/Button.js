import React from 'react';
import PropTypes from 'prop-types';
import { StyledButton } from './styled';

const Button = ({children, ...props}) => (
  <StyledButton
    {...props}
  >
    {children}
  </StyledButton>
);

Button.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.string
  ]).isRequired,
  onClick: PropTypes.func,
  small: PropTypes.bool,
  disabled: PropTypes.bool
};

Button.defaultProps = {
  disabled: false
};

export default Button;
