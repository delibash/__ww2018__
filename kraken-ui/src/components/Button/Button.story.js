import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Button from './Button';

storiesOf('Button', module)
  .add('Default Button', () => (
    <Button onClick={action('button-click')}>Button</Button>
  ))
  .add('Disabled Button', () => (
    <Button disabled={true} onClick={action('button-click')}>Disabled Button</Button>
  ))


