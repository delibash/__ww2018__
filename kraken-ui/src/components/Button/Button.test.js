import React from 'react';
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import Button from './Button';

test('Default Button', () => {
  const tree = renderer.create(<Button>Button</Button>).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('cursor', 'pointer');
  expect(tree).toHaveStyleRule('border', 'none');
});