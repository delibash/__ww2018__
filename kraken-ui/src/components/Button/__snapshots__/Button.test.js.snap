// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Button 1`] = `
.c0 {
  padding: .8rem 1.8rem;
  text-transform: capitalize;
  font-weight: 300;
  font-size: 1.3rem;
  border: none;
  border-radius: .2rem;
  background-color: rgba(100,107,255,1);
  color: white;
  cursor: pointer;
}

.c0:hover {
  background-color: rgba(234,236,242,1);
  color: rgba(0,0,0,1);
}

<button
  className="c0"
  disabled={false}
>
  Button
</button>
`;
